rasa/utils/tensorflow/temp_keras_modules.py

```diff
@@ -362,7 +362,7 @@ def fit(
             self.stop_training = False
             self.train_function = self.make_train_function()
             self._train_counter.assign(0)
-            callbacks.on_train_begin()
+            cast(callbacks_module.CallbackList, callbacks).on_train_begin()
```

Comment:
for readability I would move this below line 360 (`callbacks = cast(callbacks_module.CallbackList, callbacks)`)

Reply:
and that way you don't need all the cast() below